hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java

```diff
@@ -828,6 +828,10 @@ public Object putFromLoad(Data key, Object value) {
 
     @Override
     public Object putFromLoad(Data key, Object value, long ttl) {
+        if (key == null || value == null) {
```

Comment:
In case of doubt it is best to fail fast; one of our architectural principles. So I would also go for an exception (that isn't gobbled).